rpc.go

```diff
@@ -312,3 +310,153 @@ func getMiddlewareResult(u *User, line string) string {
 	}
 	return line
 }
+
+func pluginsCMD(_ string, u *User) {
+	plugins := make([]CMD, 0, len(PluginCMDs))
+	for n, c := range PluginCMDs {
+		plugins = append(plugins, CMD{
+			name:     n,
+			info:     c.info,
+			argsInfo: c.argsInfo,
+		})
+	}
+	autogenerated := autogenCommands(plugins)
+	if autogenerated == "" {
+		autogenerated = "   (no plugin commands are loaded)"
+	}
+	u.room.broadcast("", "Plugin commands  \n"+autogenerated)
+}
+
+type tokensDbEntry struct {
+	Token string `json:"token"`
+	Data  string `json:"data"`
+}
+
+var Tokens = make([]tokensDbEntry, 0)
+
+func initTokens() {
+	if Integrations.RPC == nil {
+		return
+	}
+
+	f, err := os.Open(Config.DataDir + string(os.PathSeparator) + "tokens.json")
+	if err != nil {
+		if !os.IsNotExist(err) {
+			Log.Fatal("Error reading tokens file: " + err.Error())
+		}
+		return
+	}
+	defer f.Close()
+	err = json.NewDecoder(f).Decode(&Tokens)
+	if err != nil {
+		error := "Error decoding tokens file: " + err.Error()
+		MainRoom.broadcast(Devbot, error)
+		Log.Fatal(error)
+	}
+}
+
+func saveTokens() {
+	f, err := os.Create(Config.DataDir + string(os.PathSeparator) + "tokens.json")
+	if err != nil {
+		Log.Fatal("Error saving tokens file: " + err.Error())
+	}
+	defer f.Close()
+	data, err := json.Marshal(Tokens)
+	if err != nil {
+		error := "Error encoding tokens file: " + err.Error()
+		MainRoom.broadcast(Devbot, error)
+		Log.Fatal(error)
+	}
+	_, err = f.Write(data)
+	if err != nil {
+		error := "Error writing tokens file: " + err.Error()
+		MainRoom.broadcast(Devbot, error)
+		Log.Fatal(error)
+	}
+}
+
+func checkToken(token string) bool {
+	if token == Integrations.RPC.Key {
```

Comment:
We can remove `Integrations.RPC.Key` completely now (tokens are better)

Reply:
Do you think it's a good idea to also have something that can be configured by the server owner that no one can revoke or view? I initially removed it but I added it back in thinking that could be useful (also to set the auth token on a plugin before you have your devzat instance deployed)

Reply:
Oh yes, good point, that could be useful. We can keep it for now.